ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -154,18 +121,23 @@ private OperationController getOperationController() {
    * @param blobId The ID of the blob for which blob data is requested.
    * @param options The options associated with the request.
    * @param callback The callback which will be invoked on the completion of the request.
-   * @return A future that would contain a {@link ReadableStreamChannel} that represents the blob data eventually.
+   * @return A future that would eventually contain a {@link GetBlobResult} that can contain either
+   *         the {@link BlobInfo}, the {@link ReadableStreamChannel} containing the blob data, or both.
    */
   @Override
-  public Future<ReadableStreamChannel> getBlob(String blobId, GetBlobOptions options,
-      Callback<ReadableStreamChannel> callback) {
+  public Future<GetBlobResult> getBlob(String blobId, GetBlobOptions options, Callback<GetBlobResult> callback) {
+    options = options == null ? GetBlobOptions.DEFAULT_OPTIONS : options;
```

Comment:
I would prefer passing a valid `GetBlobOptions` mandatory. The caller must know what type of operation it wants to make via the `getBlob()` call. Unlike the range (which is a bit fancy), it does not feel right for the router to assume the type.